code/entities/projector/FakeBounceLight.cs

```diff
@@ -0,0 +1,151 @@
+    /// 2 - Displays all previously mentioned info and also the contents of the various textures 
+    /// used by the shaders.
+    /// </summary>
+    [ConVar.Client("projector.bounce.debug")]
+    public static int BounceDebug { get; set; } = 0;
+    // You won't actually see this update in-game until issue #3288 is resolved.
+    // For more details: https://github.com/sboxgame/issues/issues/3288
+    /// <summary>
+    /// Determines the size of the texture used as the light cookie of <c>BounceSpotlight</c>.
+    /// A higher resolution may be more temporally stable, preventing some light flickering, but 
+    /// also being less realistic.
+    /// </summary>
+
+    [ConVar.Client("projector.bounce.cookiesize")]
+    public static int BounceLightCookieSize { get; set; } = 32;
+    // We use this instance field to check if the cookie size has changed since the last frame.
+    private int _previousLightCookieSize = 32;
+    /// <summary>
+    /// A factor that shall be applied at the end of brightness calculations to make the bounce
+    /// light brighter or dimmer.
+    /// </summary>
+    [ConVar.Client("projector.bounce.brightnessfactor")]
+    public static float BounceLightBrightnessFactor { get; set; } = 1f;
+    /// <summary>
+    /// Used to set the <c>FadeDistanceMin</c> of <c>BounceSpotlight</c>.
+    /// </summary>
+    [ConVar.Client("projector.bounce.fadedistancemin")]
+    public static float FadeDistanceMin { get; set; } = 1500f;
+    /// <summary>
+    /// Used to set the <c>FadeDistanceMax</c> of <c>BounceSpotlight</c>.
+    /// </summary>
+    [ConVar.Client("projector.bounce.fadedistancemax")]
+    public static float FadeDistanceMax { get; set; } = 2000f;
+    
+    /// <summary>
+    /// The texture that shall be copied from when rendering the bounce light. 
+    /// Generally, this should be the <c>LightCookie</c> of a light that represents
+    /// the projector.
+    /// </summary>
+    public Texture SourceTexture { get; set; }
+    /// <summary>
+    /// The light that projects the <c>BounceLightCookie</c> away from the screen to create the
+    /// fake bounce light effect.
+    /// </summary>
+    public SpotLightEntity BounceSpotlight { get; private set; }
+    /// <summary>
+    /// The distance between <c>Entity</c> and the screen on to which it projects.
+    /// </summary>
+    public float ScreenDistanceFromProjector { get; private set; }
+
+    private float BaseBounceLightBrightness { get; init; } = 10f;
+
+    public FakeBounceLight()
+    {
+        InitRendering();
+    }
+
+    protected override void OnActivate()
+    {
+        base.OnActivate();
+
+        BounceSpotlight ??= CreateSpotlight();
+    }
+
+    private SpotLightEntity CreateSpotlight()
+    {
+        var spotlight = new SpotLightEntity()
+        {
+            Brightness = CalculateBounceBrightness(),
+            Transform = Entity.Transform,
+            Range = 1000,
```

Comment:
Do we want to expose this to hammer? Will this range every really matter?

Reply:
I'm inclined to leave it as-is since the effect I think we'd like is that the first few rows of an audience receive light while the back rows are darker.